File: filter/filter-yml/product-api/src/main/java/project/gateway/product/core/web/presentation/ProductSearchAPI.java
package project.gateway.product.core.web.presentation;

import lombok.RequiredArgsConstructor;
import lombok.extern.slf4j.Slf4j;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import project.gateway.product.common.response.ApiResponse;
import project.gateway.product.core.web.dto.ProductResponse;

import static project.gateway.product.common.codeandmessage.common.SuccessCodeAndMessage.OK;

@Slf4j
@RestController
@RequiredArgsConstructor
@RequestMapping("/api/products")
public class ProductSearchAPI {

    @GetMapping("/{productId}")
    public ApiResponse<ProductResponse> findProductById(
            @PathVariable("productId") Long productId
    ) {
        log.info("[Product/Presentation]----xx> Search Request: {}", productId);
        return ApiResponse.of(OK, new ProductResponse(1L));
    }
}

File: load-balancer/product-api-2/src/main/kotlin/project/study/product/core/web/presentation/ProductSearchAPI.kt
package project.study.product.core.web.presentation

import lombok.extern.slf4j.Slf4j
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@Slf4j
@RestController
@RequestMapping("/api/products")
class ProductSearchAPI {

    val log: Logger = LoggerFactory.getLogger(ProductSearchAPI::class.java)

    @GetMapping("/{productId}")
    fun findProductById(@PathVariable("productId") productId: Long): ResponseEntity<ProductResponse> {
        log.info("----x> SERVER-2")
        return ResponseEntity.ok(ProductResponse(1L, "퍼즐"))
    }
}

data class ProductResponse(
    val productId: Long,
    val name: String
)
